File: fitness-backend/users-service/src/main/java/org/mwdziak/domain/Ingredient.java
package org.mwdziak.domain;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Ingredient {
    @Id
    @GeneratedValue
    private Integer id;
    private Integer fdcId;
    private String foodKind;
    private String description;
    private Double weight;
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name="nutrients_id")
    private Nutrients nutrients;
    @ManyToOne()
    @JoinColumn(name="meal_id")
    private Meal meal;
}

File: fitnessmobileclient/app/src/main/java/com/mwdziak/fitness_mobile_client/domain/Ingredient.kt
package com.mwdziak.fitness_mobile_client.domain

data class Ingredient(
    val id: Int = 0,
    var foodKind: String = "",
    var description: String = "",
    var weight: Double = 0.0,
    var unit: String = "",
    val calories: Double = 0.0,
    val protein: Double = 0.0,
    val carbohydrates: Double = 0.0,
    val fat: Double = 0.0
)